packages/test/test-end-to-end-tests/src/test/gc/gcStats.spec.ts

```diff
@@ -121,6 +121,7 @@ describeFullCompat("Garbage Collection Stats", (getTestObjectProvider) => {
         // Upload 2 attachment blobs and store their handles to mark them referenced.
         const blob1Contents = "Blob contents 1";
         const blob2Contents = "Blob contents 2";
+        await provider.ensureSynchronized();
```

Comment:
Why is this required? What changed?

Reply:
Runtime will now return blob handles while disconnected and add a new local ID to the redirect table, changing the stats. So we have to make sure we are connected before uploading the blobs.

Reply:
ensureSynchronized doesn't wait for connected state. It may or may not happen because of the different waits. You should explicitly wait for connected state. See `ensureContainerConnected` that is added to some tests.